catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -161,6 +160,29 @@ private static double metersToDegrees(double distance) {
     }
 
     @Override
+    public SolrQuery propertyIsEqualTo(String functionName, List<Object> arguments,
+            Object literal) {
+        String not;
+        SolrQuery query;
+        switch (functionName) {
+        case "divisibleBy":
+
+            //the return type is boolean so cast the literal to boolean and in effect this is just a NOT so we will update the query as such9
+            not = (Boolean) literal ? "" : "!";
+            query = propertyIsDivisibleBy((String) arguments.get(0), (Long) arguments.get(1));
+            return query.setQuery(not + query.getQuery());
+        case "proximity":
+            not = (Boolean) literal ? "" : "!";
+            query = propertyIsInProximityTo((String) arguments.get(0),
+                    (Integer) arguments.get(1),
+                    (String) arguments.get(2));
+            return query.setQuery(not + query.getQuery());
```

Comment:
I guess thats a question do I write the code so that someone could easily add a new function or since they happen to share logic now that is common to every statement currently in the switch do i drop that logic out?